### PR TITLE
SAK-50803 Sitestats "Reports Add" page shows detach warnings

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/components/StylableSelectOptionsGroup.java [hunk 1]
```diff
@@ -37,7 +42,30 @@ public StylableSelectOptionsGroup(String id, IModel model, IModel style, IModel<
...
+		}
+		// Detach the body container
+		detachBodyContainer();
+	}
+
+	private void detachBodyContainer() {
+		if (getBodyContainer() != null) {
+			getBodyContainer().detach();
+			// after detachment remove the body container so we don't process it again
+			//remove(getBodyContainer());
```
Comment (stetsche) [inline sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/components/StylableSelectOptionsGroup.java:68]:
```suggestion
```

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/components/StylableSelectOptionsGroup.java [hunk 2]
```diff
@@ -37,7 +42,30 @@ public StylableSelectOptionsGroup(String id, IModel model, IModel style, IModel<
 			optgroup.add(new AttributeModifier("class", hclass));
 		}
 		addToBorder(optgroup);
-		optgroup.add(getBodyContainer());
 	}
 
+	@Override
+	public void onComponentTagBody(MarkupStream markupStream, org.apache.wicket.markup.ComponentTag openTag) {
```
Comment (stetsche) [inline sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/components/StylableSelectOptionsGroup.java:48]: Could we `import org.apache.wicket.markup.ComponentTag`?

**Review by stetsche — CHANGES_REQUESTED** [review]
Nice!

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/components/StylableSelectOptionsGroup.java [hunk 3]
```diff
@@ -19,16 +19,21 @@
 package org.sakaiproject.sitestats.tool.wicket.components;
 
 import org.apache.wicket.AttributeModifier;
+import org.apache.wicket.markup.MarkupStream;
```
Comment (ern) [inline sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/components/StylableSelectOptionsGroup.java:22]:
```suggestion
import org.apache.wicket.markup.ComponentTag;
import org.apache.wicket.markup.MarkupStream;
```

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportDataPage.java [hunk 4]
```diff
@@ -102,7 +102,7 @@ public class ReportDataPage extends BasePage {
 	private int							selectedHeight		= 0;
 
 	// namespace for sakai icons see _icons.scss
-	public static final String ICON_SAKAI = "icon-sakai--";
+	public static final String ICON_SAKAI = "si-";
```
[the remark this answers is not in the document]
Comment (ern) [reply sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportDataPage.java:105]: 'si-' looks correct vs 'tool-'

File: sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java [hunk 5]
```diff
@@ -124,21 +124,6 @@ public class ReportsEditPage extends BasePage {
 
 	private ZonedDateTime startDate, endDate;
 
-	// namespace for sakai icons see _icons.scss
-	public static final String ICON_SAKAI = "icon-sakai--";
```
Comment (ern) [inline sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java:128]: would change this to 'si-' and remove the changes below referencing ICON-SAKAI

Comment (ern) [reply sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/wicket/pages/ReportsEditPage.java:128]:
```suggestion
	// namespace for sakai icons see _icons.scss
	public static final String ICON_SAKAI = "si-";
```
